From 026459907fbe609f2be9b52e043a607014da5651 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kamil=20Go=C5=82da?= <kamil.golda@protonmail.com>
Date: Sat, 28 Jun 2025 11:55:11 +0200
Subject: [PATCH 18/20] arm64: dts: qcom: sm6125-xiaomi-laurel-sprout: Enable
 GPU
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Enable the gpu node and provide firmware path

Signed-off-by: Kamil Gołda <kamil.golda@protonmail.com>
[rebase, enable all gpu related nodes]
Signed-off-by: Barnabás Czémán <barnabas.czeman@mainlining.org>
---
 .../dts/qcom/sm6125-xiaomi-laurel-sprout.dts  | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts b/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
index 910a7c086..8b62625e7 100644
--- a/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
+++ b/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
@@ -157,6 +157,30 @@ active-config0 {
 	};
 };
 
+&dispcc {
+	status = "okay";
+};
+
+&adreno_smmu {
+	status = "okay";
+};
+
+&gmu_wrapper {
+	status = "okay";
+};
+
+&gpu {
+	status = "okay";
+
+	zap-shader {
+		firmware-name = "qcom/sm6125/xiaomi/laurel/a610_zap.mbn";
+	};
+};
+
+&gpucc {
+	status = "okay";
+};
+
 &hsusb_phy1 {
 	vdd-supply = <&vreg_l7a>;
 	vdda-pll-supply = <&vreg_l10a>;
-- 
2.56.0

